src/d3a/d3a_core/simulation.py

```diff
@@ -265,10 +265,12 @@ def _update_and_send_results(self):
             self.file_stats_endpoint(self.area)
             return
 
-        results = self.endpoint_buffer.prepare_results_for_publish()
-        if results is None:
-            return
-        self.kafka_connection.publish(results, self._simulation_id)
+        # don't send results to kafka when running on cli even though kafka connection is enabled:
+        if not self._started_from_cli and self.kafka_connection.is_enabled():
```

Comment:
```suggestion
        if not self._started_from_cli and self.kafka_connection.is_enabled:
```
How about making `is_enabled` a property method?

Reply:
I do not really recall why, but I think that I stumbled upon issues while trying to implement exactly this (probably with unit tests and mocking the property IIRC )